src/analysis/normed_space/multilinear.lean

```diff
@@ -0,0 +1,869 @@
+/-
+Copyright (c) 2020 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+
+import analysis.normed_space.operator_norm topology.algebra.multilinear
+
+/-!
+# Operator norm on the space of continuous multilinear maps
+
+When `f` is a continuous multilinear map in finitely many variables, we define its norm `∥f∥` as the
+smallest number such that `∥f m∥ ≤ ∥f∥ * univ.prod (λi, ∥m i∥)` for all `m`.
+
+We show that it is indeed a norm, and prove its basic properties.
+
+## Main results
+
+Let `f` be a multilinear map in finitely many variables.
+* `exists_bound_of_continuous` asserts that, if `f` is continuous, then there exists `C > 0`
+  with `∥f m∥ ≤ C * univ.prod (λi, ∥m i∥)` for all `m`.
+* `continuous_of_bound`, conversely, asserts that this bound implies continuity.
+* `mk_continuous` constructs the associated continuous multilinear map.
+
+Let `f` be a continuous multilinear map in finitely many variables.
+* `∥f∥` is its norm, i.e., the smallest number such that `∥f m∥ ≤ ∥f∥ * univ.prod (λi, ∥m i∥)` for
+  all `m`.
+* `le_op_norm f m` asserts the fundamental inequality `∥f m∥ ≤ ∥f∥ * univ.prod (λi, ∥m i∥)`.
+* `norm_image_sub_le_of_bound f m₁ m₂` gives a control of the difference `f m₁ - f m₂` in terms of
+  `∥f∥` and `∥m₁ - m₂∥`.
+
+We also register isomorphisms corresponding to currying or uncurrying variables, transforming a
+continuous multilinear function `f` in `n+1` variables into a continuous linear function taking
+values in continuous multilinear functions in `n` variables, and also into a continuous multilinear
+function in `n` variables taking values in continuous linear functions. These operations are called
+`f.curry_left` and `f.curry_right` respectively (with inverses `f.uncurry_left` and
+`f.uncurry_right`). They induce continuous linear equivalences between spaces of
+continuous multilinear functions in `n+1` variables and spaces of continuous linear functions into
+continuous multilinear functions in `n` variables (resp. continuous multilinear functions in `n`
+variables taking values in continuous linear functions), called respectively
+`continuous_multilinear_curry_left_equiv` and `continuous_multilinear_curry_right_equiv`.
+
+## Implementation notes
+
+We mostly follow the API (and the proofs) of `operator_norm.lean`, with the additional complexity
+that we should deal with multilinear maps in several variables. The currying/uncurrying
+constructions are based on those in `multilinear.lean`.
```

Comment:
Could you please shortly explain why it's difficult to reuse results from `operator_norm` here? I mean, my first thought was "why not define the norm by induction as the norm of `f.curry_left`?" As far as I understand, this is possible in the case of `fin n` but causes troubles in the general case because you have to choose an order on `ι`.

Reply:
Done